### ISSUE TITLE
Unable to round-trip a DataFrame

### ISSUE THREAD

**Issue by dmbates** [JuliaData/Arrow.jl#24]
I am not sure what it is about this structure that causes the round-trip to fail.  My actual case involves the same structure and names.  I am just simulating data here for convenience.
```julia
julia> tt = DataFrame(C=rand(40), T1=rand(40), T2=rand(40), R1=rand(40), R2=rand(40), Trial = PooledArray(repeat(string.(1:4), inner=10)));

julia> Arrow.write("tt.arrow", tt)
"tt.arrow"

julia> ttnew = Arrow.Table("tt.arrow")
Arrow.Table: Error showing value of type Arrow.Table:
ERROR: KeyError: key :C not found
Stacktrace:
 [1] getindex at ./dict.jl:467 [inlined]
 [2] getcolumn at /home/bates/.julia/packages/Arrow/uVYhe/src/table.jl:38 [inlined]
 [3] #1 at ./none:0 [inlined]
 [4] iterate at ./generator.jl:47 [inlined]
 [5] collect(::Base.Generator{Array{Symbol,1},Tables.var"#1#2"{Arrow.Table}}) at ./array.jl:686
 [6] _totuple at ./tuple.jl:258 [inlined]
 [7] Tuple at ./tuple.jl:230 [inlined]
 [8] NamedTuple(::Arrow.Table) at /home/bates/.julia/packages/Tables/pKMcn/src/Tables.jl:177
 [9] show(::IOContext{REPL.Terminals.TTYTerminal}, ::Arrow.Table) at /home/bates/.julia/packages/Tables/pKMcn/src/Tables.jl:183
 [10] show(::IOContext{REPL.Terminals.TTYTerminal}, ::MIME{Symbol("text/plain")}, ::Arrow.Table) at ./multimedia.jl:47
 [11] display(::REPL.REPLDisplay, ::MIME{Symbol("text/plain")}, ::Any) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/REPL/src/REPL.jl:214
 [12] display(::REPL.REPLDisplay, ::Any) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/REPL/src/REPL.jl:218
 [13] display(::Any) at ./multimedia.jl:328
 [14] #invokelatest#1 at ./essentials.jl:710 [inlined]
 [15] invokelatest at ./essentials.jl:709 [inlined]
 [16] print_response(::IO, ::Any, ::Bool, ::Bool, ::Any) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/REPL/src/REPL.jl:238
 [17] print_response(::REPL.AbstractREPL, ::Any, ::Bool, ::Bool) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/REPL/src/REPL.jl:223
 [18] (::REPL.var"#do_respond#54"{Bool,Bool,REPL.var"#64#73"{REPL.LineEditREPL,REPL.REPLHistoryProvider},REPL.LineEditREPL,REPL.LineEdit.Prompt})(::Any, ::Any, ::Any) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/REPL/src/REPL.jl:822
 [19] #invokelatest#1 at ./essentials.jl:710 [inlined]
 [20] invokelatest at ./essentials.jl:709 [inlined]
 [21] run_interface(::REPL.Terminals.TextTerminal, ::REPL.LineEdit.ModalInterface, ::REPL.LineEdit.MIState) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/REPL/src/LineEdit.jl:2355
 [22] run_frontend(::REPL.LineEditREPL, ::REPL.REPLBackendRef) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/REPL/src/REPL.jl:1144
 [23] (::REPL.var"#38#42"{REPL.LineEditREPL,REPL.REPLBackendRef})() at ./task.jl:356
```

**Comment by dmbates** [comment]
@quinnj Are you able to reproduce this problem?  Please let me know if there is anything further I can do to clarify.

**Comment by quinnj** [comment]
I can reproduce; I'll dig in.

**Comment by dmbates** [comment]
I tried out the round-trip test on the jq/compression branch and it works fine.  It still produces an error on the original file that gave me the problem, https://osf.io/xy5at/download, so it may be something in the writing of the file rather than the reading.  However, it is not a problem to recreate the file so this issue can be closed by #26 I think.

**Comment by quinnj** [comment]
@dmbates, are you able to read that file anywhere? It seems like there's corrupt data or something to me; and I can't manage to read it with `pyarrow` either:
```python
>>> file = "/Users/jacobquinn/Downloads/sdio_relaxivity.arrow"
>>> buf = file
>>> pa.ipc.open_file(buf).read_pandas()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.8/site-packages/pyarrow/ipc.py", line 49, in read_pandas
    table = self.read_all()
  File "pyarrow/ipc.pxi", line 556, in pyarrow.lib._RecordBatchFileReader.read_all
  File "pyarrow/error.pxi", line 122, in pyarrow.lib.pyarrow_internal_check_status
  File "pyarrow/error.pxi", line 84, in pyarrow.lib.check_status
pyarrow.lib.ArrowInvalid: First or last binary offset out of bounds
>>> reader = pa.ipc.open_file(buf)
>>> reader
<pyarrow.ipc.RecordBatchFileReader object at 0x123201ca0>
>>> reader.
reader.get_batch(          reader.get_record_batch(   reader.num_record_batches  reader.read_all(           reader.read_pandas(        reader.schema
>>> reader.num_record_batches
1
>>> reader.schema
C: float not null
T1: double not null
T2: double not null
R1: double not null
R2: double not null
Trial: dictionary<values=string, indices=uint8, ordered=0> not null
>>> reader.read_all()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "pyarrow/ipc.pxi", line 556, in pyarrow.lib._RecordBatchFileReader.read_all
  File "pyarrow/error.pxi", line 122, in pyarrow.lib.pyarrow_internal_check_status
  File "pyarrow/error.pxi", line 84, in pyarrow.lib.check_status
pyarrow.lib.ArrowInvalid: First or last binary offset out of bounds
>>>
```
